hedera-node/hedera-app/src/main/java/com/hedera/node/app/tss/TssBaseServiceImpl.java

```diff
@@ -146,27 +148,34 @@ public void setCandidateRoster(@NonNull final Roster roster, @NonNull final Hand
         final var candidateRosterParticipantDirectory = computeTssParticipantDirectory(roster, maxSharesPerNode, (int)
                 context.networkInfo().selfNodeInfo().nodeId());
 
-        final AtomicInteger shareIndex = new AtomicInteger(0);
-        for (final var tssPrivateShare : tssPrivateShares) {
-            final var tssMsg = CompletableFuture.supplyAsync(
-                            () -> tssLibrary.generateTssMessage(candidateRosterParticipantDirectory, tssPrivateShare),
-                            tssLibraryExecutor)
-                    .exceptionally(e -> {
-                        log.error("Error generating tssMessage", e);
-                        return null;
-                    });
-            tssMsg.thenAccept(msg -> {
-                if (msg == null) {
-                    return;
-                }
-                final var tssMessage = TssMessageTransactionBody.newBuilder()
-                        .sourceRosterHash(activeRosterHash)
-                        .targetRosterHash(candidateRosterHash)
-                        .shareIndex(shareIndex.getAndAdd(1))
-                        .tssMessage(Bytes.wrap(msg.bytes()))
-                        .build();
-                tssSubmissions.submitTssMessage(tssMessage, context);
-            });
+        // https://github.com/hashgraph/hedera-services/issues/16166
+        final boolean hasKeyMaterial = false;
```

Comment:
I thought we are opening this PR only for happy path. This is always true right ?

Reply:
My mistake, this should be `true` for the happy path.

I asked yesterday, this is _not_ always true. Key material can be missing from an _active_ roster IFF there is no existing ledger Id, i.e. in the genesis or new network scenarios. So it's _almost_ always true.